### PR TITLE
Improve docs around streaming pso desc

### PR DESCRIPTION
**Pull request by john-h-k**
Describe the requirement and layout of a PSO desc, and document all the subobjects to their matching subobject_type enum members

### CODE REVIEW

**Comment by PRMerger19** [comment]
@john-h-k : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: sdk-api-src/content/d3d12/ns-d3d12-d3d12_pipeline_state_stream_desc.md [hunk 1]
```diff
@@ -75,9 +76,39 @@ Specifies the address of a data structure that describes as a bytestream an arbi
...
+struct alignas(void*) StreamingRasterizerDesc
+{
+private:
+  D3D12_PIPELINE_STATE_SUBOBJECT_TYPE Type = D3D12_PIPELINE_STATE_SUBOBJECT_TYPE_RASTERIZER;
+public:
+  D3D12_RASTERIZER_DESC Desc;
+}
+```
+
+The runtime will determine the type of a pipeline stream (valid types being **COMPUTE**, **GRAPHICS**, and **MESH**), by which subobject type, out of **VS** (vertex shader), **CS** (compute shader), and **MS** (mesh shader), is found first. If the runtime finds none of these shaders, it will fail pipeline creation.
```
Comment (jenatali) [inline sdk-api-src/content/d3d12/ns-d3d12-d3d12_pipeline_state_stream_desc.md]: I'd leave out first. If multiple types are found, creation also fails.

**Review by jenatali — APPROVED** [review]
Looks great, thanks for writing that up!

**Comment by PRMerger10** [comment]
@john-h-k : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger12** [comment]
@stevewhims : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by stevewhims** [comment]
It looks like something went wrong with this PR, and nothing actually changed when I completed it. So I've recreated the changes here in a new PR: https://cpubwin.visualstudio.com/win32/_git/sdk-api/pullrequest/17255.

**Pull request merged**